compiler/rustc_attr_parsing/src/parser.rs

```diff
@@ -72,7 +72,8 @@ impl<'a> PathParser<'a> {
     /// Unlike [`segments_is`](Self::segments_is),
     /// `self` may contain more segments than the number matched  against.
     pub fn starts_with(&self, segments: &[Symbol]) -> bool {
-        segments.len() < self.len() && self.segments().zip(segments).all(|(a, b)| a.name == *b)
+        segments.len() < self.len()
```

Comment:
is this really better than what was there before? you still need to take() after the less-than check